src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs

```diff
@@ -42,29 +42,28 @@ public Task PostValidateOptions()
         [RavenAction("/databases/*/smuggler/export", "POST")]
         public async Task PostExport()
         {
-            DocumentsOperationContext context;
+            DocumentsOperationContext context;            
             using (ContextPool.AllocateOperationContext(out context))
             using (context.OpenReadTransaction())
             {
                 var operationId = GetIntValueQueryString("operationId", required: false);
 
-                var exporter = new SmugglerExporter(Database)
-                {
-                    DocumentsLimit = GetIntValueQueryString("documentsLimit", required: false),
-                    RevisionDocumentsLimit = GetIntValueQueryString("RevisionDocumentsLimit", required: false),
-                };
-
-                var operateOnTypes = GetStringQueryString("operateOnTypes", required: false);
-                DatabaseItemType databaseItemType;
-                if (Enum.TryParse(operateOnTypes, true, out databaseItemType))
-                {
-                    exporter.OperateOnTypes = databaseItemType;
-                }
+                var blittableJson = await context.ReadForMemoryAsync(RequestFormStream("DownloadOptions"), "DownloadOptions");
+                var options = JsonDeserializationServer.DatabaseExportOptions(blittableJson);
 
+                var exporter = new SmugglerExporter(Database, options);
                 var token = CreateOperationToken();
 
+                var fileName = string.IsNullOrEmpty(exporter.Options.FileName) ?
```

Comment:
Detect if the file contains any invalid file name chars (check 3.5 - StudioTasksController)